chore: optimise Date object creation in graph components and utilities

joeljkrb opened the pull request:
- Add useMemo to PayoutBar and AveragePayoutLine to cache current date
- Memo-ise date calculations in ActiveGraph and Payouts components
- Implement caching mechanism in dateUtils.ts with 1-minute TTL
- Cache timestamp in useEraTimeLeft hook with 1-second TTL
- Reduce memory allocation and improve render performance

joeljkrb committed:
chore: optimise Date object creation in graph components and utilities

- Add useMemo to PayoutBar and AveragePayoutLine to cache current date
- Memo-ise date calculations in ActiveGraph and Payouts components
- Implement caching mechanism in dateUtils.ts with 1-minute TTL
- Cache timestamp in useEraTimeLeft hook with 1-second TTL
- Reduce memory allocation and improve render performance

joeljkrb committed:
Merge branch 'main' into jk-dateobject

rossbulat commented on packages/app/src/hooks/useEraTimeLeft/index.tsx:
```diff
@@ -25,8 +39,8 @@ export const useEraTimeLeft = () => {
 
-    // Estimate remaining time of era
-    const timeleft = Math.max(0, end - getUnixTime(new Date()))
+    // Estimate remaining time of era using cached timestamp
```
Because this is a getter fn, `new Date` is only executed when the consuming component calls it, so I don't think this caching will make a difference

joeljkrb committed:
Merge branch 'main' into jk-dateobject

joeljkrb committed:
refactor: remove ineffective date optimisations

- Remove unnecessary useMemo in ActiveGraph.tsx, replace with date-fns functions
- Remove ineffective 1-second caching in useEraTimeLeft hook
- Keep dateUtils.ts caching as it may benefit future usage patterns

joeljkrb committed:
Merge branch 'jk-dateobject' of https://github.com/polkadot-cloud/polkadot-staking-dashboard into jk-dateobject

joeljkrb committed:
Merge branch 'main' into jk-dateobject

joeljkrb committed:
Merge branch 'main' into jk-dateobject

joeljkrb committed:
Merge branch 'main' into jk-dateobject

rossbulat commented:
Resolved in #3137
